lib/ddtrace/transport/trace_formatter.rb

```diff
@@ -59,7 +59,7 @@ def set_resource!
         # If the trace resource is undefined, or the root span wasn't
         # specified, don't set this. We don't want to overwrite the
         # resource of a span that is in the middle of the trace.
-        return if trace.resource.nil? || !@found_root_span
+        return if trace.resource.nil? || partial?
```

Comment:
`@found_root_span` kind of comes from `TraceSegment.root_span_id`, which comes from voluntarily not passing `root_span.id` in `TraceOperation` when it is known `partial` (from `TraceOperation#finished?`), so `@found_root_span` seems like a roundabout way to infer that this is a partial trace.

I wonder if we should just pass the `finished?`/`partial` information via `TraceSegment` instead of relying on second-order divination.

In fact, it wasn't clear to me whether `resource` being `nil` was supposed to be part of that `partial?` predicate or not, but either way it does feel like divination from external behaviour.